src/pytest_netdut/wrappers.py

```diff
@@ -34,36 +34,84 @@ def _splitcmds(cmds):
     return cmds
 
 
-def _eos_to_mos_translator(cmds):
-    translations = [
+class Translator:
+    """Base class which implements translation functions for hiding syntax
+    and result differences between devices with different operating systems.
+
+    Override the standardize_key function to implement result processing.
+
+    Args:
+        config_patterns (iterable[regex pattern]): List of regex patterns for CLI translations.
+    """
+
+    config_patterns = []
+
+    def standardize_key(self, k):
+        """Returns a standardized key.
+
+        Args:
+            k (str): Key to process.
+        """
+        raise NotImplementedError
+
+    def process_dict(self, d):
+        result = {}
+        for k, v in d.items():
+            nk = self.standardize_key(str(k))
+            result[nk] = self.process_dict(v) if isinstance(v, dict) else v
+        return result
+
+    def cli(self, cmds):
+        """CLI translator.
+
+        Args:
+            cmds (str | Iterable[str]): Either an iterable of commands to be run or a newline separated string containing one or more commands.
+        """
+        commands = []
+        for command in _splitcmds(cmds):
+            for (before, after) in self.config_patterns:
+                matcher = re.match(before, command)
+                if matcher:
+                    commands.append(matcher.expand(after))
+                    break
+            else:
+                # If we don't get a match, just append the original.
+                commands.append(command)
+        if commands != cmds:
+            logging.debug("Before: %s, After: %s", repr(cmds), repr(commands))
+        return commands
+
+    def json(self, data):
+        """JSON translator.
+
+        Args:
+            data (str | dict): Results are typically a string or a dictionary.
+        """
+        if isinstance(data, dict):
+            return self.process_dict(data)
```

Comment:
Should this be called "translate_json_dict" or similar to make it more clear what this is doing?